bitbots_path_planning/bitbots_path_planning/controller.py

```diff
@@ -0,0 +1,119 @@
+        final_walk_angle = math.atan2(
+            end_pose.position.y - current_pose.position.y,
+            end_pose.position.x - current_pose.position.x)
+
+        # Calculate the distance from our current position to the final position of the global plan
+        distance = math.hypot(
+            end_pose.position.x - current_pose.position.x,
+            end_pose.position.y - current_pose.position.y)
+
+        # Calculate the translational walk velocity. It considers the distance and breaks if we are close to the final position of the global plan
+        walk_vel = min(distance * self.config_translation_slow_down_factor, self.config_max_vel_x)
+
+        # Check if we are so close to the final position of the global plan that we want to align us with its orientation and not the heading towards its position
+        diff = 0
+        if distance > self.config_orient_to_goal_distance:
+            # Calculate the difference between our current heading and the heading towards the final position of the global plan
+            diff = final_walk_angle - self._get_yaw(current_pose.orientation)
+        else:
+            # Calculate the difference between our current heading and the heading of the final position of the global plan
+            diff = self._get_yaw(end_pose.orientation) - self._get_yaw(current_pose.orientation)
+
+        # Get the min angle of the difference
+        min_angle = math.remainder(diff, math.tau)
+        # Calculate our desired rotation velocity based on the angle difference and our max velocity
+        rot_goal_vel = min(max(self.config_rotation_slow_down_factor * min_angle, -self.config_max_rotation_vel), self.config_max_rotation_vel)
+
+        # Calculate the x and y components of our linear velocity based on the desired heading and the desired translational velocity.
+        cmd_vel.linear.x = math.cos(walk_angle - self._get_yaw(current_pose.orientation)) * walk_vel
+        cmd_vel.linear.y = math.sin(walk_angle - self._get_yaw(current_pose.orientation)) * walk_vel
+
+        # Scale command accordingly if a limit is acceded
+        if cmd_vel.linear.x > self.config_max_vel_x:
+            self.node.get_logger().debug("X LIMIT reached: %f > %f, with y %f".format(cmd_vel.linear.y, self.config_max_vel_x, cmd_vel.linear.y));
+            cmd_vel.linear.y *= self.config_max_vel_x / cmd_vel.linear.x
+            cmd_vel.linear.x = self.config_max_vel_x
+            self.node.get_logger().debug("X LIMIT set y %f".format(cmd_vel.linear.y))
+
+        if cmd_vel.linear.x < self.config_min_vel_x:
+            self.node.get_logger().debug("X LIMIT reached: %f < %f, with y %f".format(cmd_vel.linear.x, self.config_min_vel_x, cmd_vel.linear.y))
+            cmd_vel.linear.y *= self.config_min_vel_x / cmd_vel.linear.x
+            cmd_vel.linear.x =self. config_min_vel_x
+            self.node.get_logger().debug("X LIMIT set y %f".format(cmd_vel.linear.y))
+
+        max_y = self.config_max_vel_y
```

Comment:
Why this local variable? Just because its name is shorter?

Reply:
Exactly

Reply:
Please don't do this. This is not explicit.